test/e2e/framework/kcp.go

```diff
@@ -411,3 +406,74 @@ func (c *kcpServer) monitorEndpoint(client *rest.RESTClient, endpoint string) {
 		}
 	}, 1*time.Second)
 }
+
+// loadKubeConfig loads a kubeconfig from disk. This method is
+// intended to be common between fixture for servers whose lifecycle
+// is test-managed and fixture for servers whose lifecycle is managed
+// separately from a test run.
+func loadKubeConfig(kubeconfigPath string) (clientcmd.ClientConfig, error) {
+	fs, err := os.Stat(kubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+	if fs.Size() == 0 {
+		return nil, fmt.Errorf("%s points to an empty file", kubeconfigPath)
+	}
+
+	rawConfig, err := clientcmd.LoadFromFile(kubeconfigPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load admin kubeconfig: %w", err)
+	}
+
+	return clientcmd.NewNonInteractiveClientConfig(*rawConfig, "root", nil, nil), nil
```

Comment:
unrelated to the this PR: "root" here is strange. That's the root workspace, i.e. not system:admin. Wondering what this client config is used for. Is it for health checks?